ucsmsdk/utils/comparesyncmo.py

```diff
@@ -0,0 +1,555 @@
+    """
+    Internal method used to translate a managed object which helps in comparing
+    two mo with different dn.
+    """
+
+    ref_dn = None
+    if xlate_org is not None:
+        match_obj = re.match(
+            r'^(org-[\-\.:_a-zA-Z0-9]{1,16}/)*org-[\-\.:_a-zA-Z0-9]{1,16}',
+            mo.dn)
+        if match_obj:
+            if mo.get_class_id() == "OrgOrg":
+                ref_dn = re.sub("%s" % (match_obj.group(0)),
+                                "%s" % xlate_org,
+                                mo.dn)
+            else:
+                ref_dn = re.sub("^%s/" % (match_obj.group(0)),
+                                "%s/" % xlate_org,
+                                mo.dn)
+            _update_mo_dn_along_with_naming_properties(mo, ref_dn)
+
+    if xlate_map is not None:
+        diff_dn = mo.dn
+        if diff_dn in xlate_map:
+            ref_dn = xlate_map[diff_dn]
+        else:
+            diff_dn = re.sub(r'[/]*[^/]+$', '', diff_dn)
+            while diff_dn is not None or diff_dn == "":
+                if diff_dn not in xlate_map:
+                    diff_dn = re.sub(r'[/]*[^/]+$', '', diff_dn)
+                    continue
+
+                ref_dn = re.sub("^%s/" % diff_dn,
+                                "%s/" % xlate_map[diff_dn],
+                                mo.dn)
+                break
+        _update_mo_dn_along_with_naming_properties(mo, ref_dn)
+
+    return mo
+
+
+def _list_has_values(obj):
+    """
+    Internal function to check if obj is non empty list.
+    """
+
+    return isinstance(obj, list) and len(obj) > 0
+
+
+def _should_skip_mo(mo):
+    """
+    Internal function to check if mo to be skipped from comparison.
+    """
+
+    if mo is None:
+        return True
+    if isinstance(mo, GenericMo):
+        if ucsgenutils.word_u(mo.get_class_id()) in skip_mos:
+            return True
+    elif mo.get_class_id() in skip_mos or \
+            mo.mo_meta.inp_out == MoMeta.ACCESS_TYPE_OUTPUTONLY or \
+            (mo.mo_meta.inp_out == MoMeta.ACCESS_TYPE_IO
+             and len(mo.mo_meta.access) == 1
+             and mo.mo_meta.access[0] == "read-only"):
+        return True
+    return False
+
+
+def _get_skip_props(mo, include_operational=False, no_version_filter=False):
+    """
+    Internal function to skip mo property if not to be considered for sync.
+    """
+
+    skip_props = []
+    for prop in mo.prop_meta:
+        mo_property_meta = mo.prop_meta[prop]
+        if mo_property_meta is None:
+            continue
+
+        # not include operational property
+        if not include_operational:
+            if mo_property_meta.access in (MoPropertyMeta.INTERNAL,
+                                           MoPropertyMeta.READ_ONLY):
+                skip_props.append(prop)
+        # checks if property is part of current or earlier ucsm schema
+        if not no_version_filter:
+            version = mo.get_handle().version
```

Comment:
I default it to None so it should not break the functionality of existing code. In compare and sync I did not put a check for handle because mo should always contain handle now and if not the compare sync should fail.